src/backends/sqlite3/session.cpp

```diff
@@ -162,10 +180,58 @@ void sqlite3_session_backend::rollback()
     execude_hardcoded(conn_, "ROLLBACK", "Cannot rollback transaction.");
 }
 
+struct SeqCtxt
+{
+    long long value_;
+    bool filledIn_;
+};
+
+static int GetOneLong(void* ctxt, int valueNum, char** values, char**)
+{
+    SeqCtxt* seqCtxt = (SeqCtxt*)ctxt;
+    seqCtxt->value_ = 0;
+    seqCtxt->filledIn_ = true;
+    char* ptr = NULL;
+    if (valueNum == 1 && values[0])
+        seqCtxt->value_ = strtol(values[0], &ptr, 10);
+    return 0;
+}
+
 bool sqlite3_session_backend::get_last_insert_id(
-    session & /* s */, std::string const & /* table */, long long & value)
+    session &, std::string const & table, long long & value)
 {
-    value = static_cast<long long>(sqlite3_last_insert_rowid(conn_));
+    char *zErrMsg = NULL;
+    SeqCtxt seqCtxt;
+    seqCtxt.filledIn_ = false;
+    seqCtxt.value_ = 0;
+    if (sequenceTableExists_ || SequenceTableExists(conn_))
+    {
+        // Once the sqlite_sequence table is created (because of a column marked AUTOINCREMENT)
+        // it can never be dropped, so don't search for it again.
+        sequenceTableExists_ = true;
+
+        std::string const query = "select seq from sqlite_sequence where name ='" + table + "'";
+        int const res = sqlite3_exec(conn_, query.c_str(), &GetOneLong, &seqCtxt, &zErrMsg);
+        check_sqlite_err(conn_, res, "Unable to get value in sqlite_sequence");
+
+        // The value will not be filled if the callback was never called.
+        // It may mean either that nothing was inserted yet into the table
+        // that has an AUTOINCREMENT column, or that the table does not have an AUTOINCREMENT
+        // column.
+        if (seqCtxt.filledIn_)
+        {
+            value = seqCtxt.value_;
+            return true;
+        }
+    }
+
+    // Fall-back just get the maximum rowid of what was already inserted in the
+    // table. This has the disadvantage that if rows were deleted, then ids may be re-used.
+    // But, if one cares about that, AUTOINCREMENT should be used anyway.
+    std::string const maxRowIdQuery = "select max(rowid) from " + table;
```

Comment:
Shouldn't table be at least quoted here?

Reply:
No, no need, the rowid is part of the table.

Reply:
Just to be clear: this is a continuation of my comment about SQL injection above, i.e. I'd prefer not to use `table` as part of the query, but if we use it, it should at least be `"'" + table + "'"`, I think.